projects/core/modules/notifications/notification-alert/notification-alert.component.ts

```diff
@@ -20,16 +26,16 @@ export class TuiNotificationAlertComponent<O, I> {
     item?: NotificationAlert<O, I>;
 
     constructor(
-        @Inject(TuiDestroyService)
-        destroy$: TuiDestroyService,
+        @Inject(ElementRef) {nativeElement}: ElementRef<HTMLElement>,
+        @Inject(TuiDestroyService) destroy$: TuiDestroyService,
     ) {
         timer(ALERT_AUTOCLOSE_TIMEOUT)
-            .pipe(takeUntil(destroy$))
-            .subscribe(() => {
-                if (this.safeItem.autoClose) {
-                    this.closeDialog();
-                }
-            });
+            .pipe(
+                takeUntil(race(destroy$, fromEvent(nativeElement, 'mouseover'))),
```

Comment:
I think this way it would cause a memory leak as it would keep on waiting in `repeatWhen` for `mouseleave` when the notification is destroyed. We probably need one `takeUntil` with `mouseenter` (btw event should be `mouseenter`) above `repeatWhen` and another with `destroy$` after.

Reply:
Yes, good catch, even when timer completes, repeatWhen is still listening.

Reply:
Done.